containers/src/Data/IntMap/Internal.hs

```diff
@@ -62,2904 +43,1113 @@
-    , merge
-    -- *** @WhenMatched@ tactics
-    , zipWithMaybeMatched
-    , zipWithMatched
-    -- *** @WhenMissing@ tactics
-    , mapMaybeMissing
-    , dropMissing
-    , preserveMissing
-    , mapMissing
-    , filterMissing
-
-    -- ** Applicative general combining function
-    , WhenMissing (..)
-    , WhenMatched (..)
-    , mergeA
-    -- *** @WhenMatched@ tactics
-    -- | The tactics described for 'merge' work for
-    -- 'mergeA' as well. Furthermore, the following
-    -- are available.
-    , zipWithMaybeAMatched
-    , zipWithAMatched
-    -- *** @WhenMissing@ tactics
-    -- | The tactics described for 'merge' work for
-    -- 'mergeA' as well. Furthermore, the following
-    -- are available.
-    , traverseMaybeMissing
-    , traverseMissing
-    , filterAMissing
-
-    -- ** Deprecated general combining function
-    , mergeWithKey
-    , mergeWithKey'
-
-    -- * Traversal
-    -- ** Map
-    , map
-    , mapWithKey
-    , traverseWithKey
-    , traverseMaybeWithKey
-    , mapAccum
-    , mapAccumWithKey
-    , mapAccumRWithKey
-    , mapKeys
-    , mapKeysWith
-    , mapKeysMonotonic
-
-    -- * Folds
-    , foldr
-    , foldl
-    , foldrWithKey
-    , foldlWithKey
-    , foldMapWithKey
-
-    -- ** Strict folds
-    , foldr'
-    , foldl'
-    , foldrWithKey'
-    , foldlWithKey'
-
-    -- * Conversion
-    , elems
-    , keys
-    , assocs
-    , keysSet
-    , fromSet
-
-    -- ** Lists
-    , toList
-    , fromList
-    , fromListWith
-    , fromListWithKey
-
-    -- ** Ordered lists
-    , toAscList
-    , toDescList
-    , fromAscList
-    , fromAscListWith
-    , fromAscListWithKey
-    , fromDistinctAscList
-
-    -- * Filter
-    , filter
-    , filterWithKey
-    , restrictKeys
-    , withoutKeys
-    , partition
-    , partitionWithKey
-
-    , mapMaybe
-    , mapMaybeWithKey
-    , mapEither
-    , mapEitherWithKey
-
-    , split
-    , splitLookup
-    , splitRoot
-
-    -- * Submap
-    , isSubmapOf, isSubmapOfBy
-    , isProperSubmapOf, isProperSubmapOfBy
-
-    -- * Min\/Max
-    , lookupMin
-    , lookupMax
-    , findMin
-    , findMax
-    , deleteMin
-    , deleteMax
-    , deleteFindMin
-    , deleteFindMax
-    , updateMin
-    , updateMax
-    , updateMinWithKey
-    , updateMaxWithKey
-    , minView
-    , maxView
-    , minViewWithKey
-    , maxViewWithKey
-
-    -- * Debugging
-    , showTree
-    , showTreeWith
-
-    -- * Internal types
-    , Mask, Prefix, Nat
-
-    -- * Utility
-    , natFromInt
-    , intFromNat
-    , link
-    , linkWithMask
-    , bin
-    , binCheckLeft
-    , binCheckRight
-    , zero
-    , nomatch
-    , match
-    , mask
-    , maskW
-    , shorter
-    , branchMask
-    , highestBitMask
-
-    -- * Used by "IntMap.Merge.Lazy" and "IntMap.Merge.Strict"
-    , mapWhenMissing
-    , mapWhenMatched
-    , lmapWhenMissing
-    , contramapFirstWhenMatched
-    , contramapSecondWhenMatched
-    , mapGentlyWhenMissing
-    , mapGentlyWhenMatched
-    ) where
```

Comment:
Can you bring back the export list? It might actually affect performance: https://www.joachim-breitner.de/blog/762-Faster_Winter_4__Export_lists

Reply:
I also like export lists, whether or not they affect performance, as documentation.